app/views/support_interface/providers/show.html.erb

```diff
@@ -2,6 +2,16 @@
 
 <h2 class='govuk-heading-l'><%= pluralize(@provider.courses.size, 'course') %> (<%= @provider.courses.visible_to_candidates.size %> on DfE Apply)</h2>
 
+<%= form_with model: @provider, url: support_interface_provider_path(@provider), method: :post do |f| %>
+  <h3 class="govuk-heading-m">
+    Open all courses
+  </h3>
+
+  <p class="govuk-body">This will toggle every course on this provider to be available through Apply.</p>
+
+  <%= f.govuk_submit "Open #{pluralize(@provider.courses.size, 'course')}" %>
+<% end %>
```

Comment:
Edge case: what if there are no courses for this provider?

Reply:
Added an if statement